### PR TITLE
[python] Add xbmcgui.DialogBusy()

### PR DESCRIPTION
**Pull request by phil65**
## Description

Adds a new dialog class to the python interface.
## Motivation and Context

Lot of python scripters are making use of our busy dialog although there is no proper interface to do that. So their solution is to use executebuiltin("ActivateWindow(busydialog)") and executebuiltin("Dialog.Close(busydialog)") in order to open / close it, which feels very workaround-ish.
Because of that, I added busydialog handling to our xbmcgui class, with a similar interface as our progress dialog.
When doin this I also found out that our busydialog core code supports a progress bar. That one can be used via python as well now, and I also added one to estuary in a separate commit.
## How Has This Been Tested?

With some simple test scripts
## Screenshots (if appropriate):

screenshot041
## Types of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by phil65** [comment]
@tamland for review, please.

File: xbmc/interfaces/legacy/Dialog.cpp [hunk 1]
```diff
@@ -491,6 +491,62 @@ namespace XBMCAddon
...
+    void DialogBusy::create()
+    {
+      DelayedCallGuard dcguard(languageHook);
+      CGUIDialogBusy* pDialog = static_cast<CGUIDialogBusy*>(g_windowManager.GetWindow(WINDOW_DIALOG_BUSY));
+
+      if (pDialog == nullptr)
+        throw WindowException("Error: Window is NULL, this is not possible :-)");
+
+      dlg = pDialog;
+      open = true;
```
Comment (tamland) [inline xbmc/interfaces/legacy/Dialog.cpp]: this open flag doesn't seem to do anything..

File: xbmc/interfaces/legacy/Dialog.cpp [hunk 2]
```diff
@@ -491,6 +491,62 @@ namespace XBMCAddon
...
+
+      pDialog->Open();
+    }
+
+    void DialogBusy::update(int percent) const
+    {
+      DelayedCallGuard dcguard(languageHook);
+      auto pDialog = dlg;
```
Comment (tamland) [inline xbmc/interfaces/legacy/Dialog.cpp]: does nothing

File: xbmc/interfaces/legacy/Dialog.h [hunk 3]
```diff
@@ -679,6 +680,131 @@ namespace XBMCAddon
...
+      ///
+      /// \ingroup python_DialogBusy
+      /// \python_func{ xbmcgui.DialogBusy().update(percent) }
+      ///------------------------------------------------------------------------
+      ///
+      /// Updates the busy dialog.
+      ///
+      /// @param percent        integer - percent complete. (0:100)
+      ///
+      /// @note If percent == 0, the progressbar will be hidden.
```
Comment (tamland) [inline xbmc/interfaces/legacy/Dialog.h]: Sounds like a bug. It should be possible to have a progress bar at 0%

File: xbmc/interfaces/legacy/Dialog.h [hunk 4]
```diff
@@ -679,6 +680,131 @@ namespace XBMCAddon
...
+      ///
+      ///
+      ///------------------------------------------------------------------------
+      ///
+      /// **Example:**
+      /// ~~~~~~~~~~~~~{.py}
+      /// ..
+      /// dialog.close()
+      /// ..
+      /// ~~~~~~~~~~~~~
```
Comment (tamland) [inline xbmc/interfaces/legacy/Dialog.h]: I think you can drop all of these "examples". They are pretty useless!:)

File: xbmc/interfaces/legacy/Dialog.h (hunk 3 above)
Comment (phil65) [inline xbmc/interfaces/legacy/Dialog.h]: Yup, I just did it in same way as progress dialog. I agree that it makes sense to change this behaviour for both.

File: xbmc/interfaces/legacy/Dialog.h (hunk 4 above)
Comment (phil65) [inline xbmc/interfaces/legacy/Dialog.h]: Ok, will only keep the one for initialization then. ( .create() )

**Comment by phil65** [comment]
@MartijnKaijser Should I add the @python_v17 stuff just to the class description or to every single method of the class?

File: xbmc/interfaces/legacy/Dialog.h [hunk 5]
```diff
@@ -744,14 +746,7 @@ namespace XBMCAddon
...
       ///
-      /// **Example:**
-      /// ~~~~~~~~~~~~~{.py}
-      /// ..
-      /// dialog.update(25)
-      /// ..
-      /// ~~~~~~~~~~~~~
-      ///
-      update(...);
+       update(...);
```
Comment (garbear) [inline xbmc/interfaces/legacy/Dialog.h]: whitespace oops

**Comment by MartijnKaijser** [comment]
Each one please as that's clearest

**Comment by MartijnKaijser** [comment]
@tamland ping for review if you have a moment

**Comment by tamland** [comment]
Need to update the documentation on the percent parameter, otherwise it looks good.

**Comment by phil65** [comment]
updated.

**Comment by MartijnKaijser** [comment]
Jenkins build this please

**Comment by FernetMenta** [comment]
@phil65 calling busy dialog from python is nonsense. the busy dialog is made to keep the GUI going while the calling thread is blocked.

calling dialogBusy from python leads to having multiple busy dialogs open at the same time:
https://trac.kodi.tv/ticket/17896

further: opening a modal dialog via a message is wrong anyway. leads to having the busy dialog open but obscured by some other dialog.

**Comment by FernetMenta** [comment]
I will deprecte this and log a message when this is called.

**Comment by FernetMenta** [comment]
> Please also make sure to block Dialog.Close(busydialog) and ActivateWindow(busydialog) then.

already had ActivateWindow(busydialog), added closedialog now

**Pull request merged**